### ISSUE TITLE
Approximately how many Serper API credits (or $) were used per training session?

### ISSUE THREAD

**Issue by minyoungbae** [GuoqingWang1/IGPO#8]

**Comment by GuoqingWang1** [comment]
Please accept my apologies, but as the Serper API is purchased by a specialized department within our company, I am unable to provide an estimate for the actual expenses.

**Comment by minyoungbae** [comment]
Thank you very much for the kind response. I ran the provided code exactly as is (train.sh -> evaluate.sh), but I couldn't reproduce the F1 score reported in the paper. Are there any differences between the current version of the code and the experimental setup used in the paper?

Below is my evaluation output, which is 5-10% lower F1 than reported:
{
    "val/test_score/2wiki_f1": 0.6205479633214008,
    "val/test_score/2wiki_f1_format_penalty": 0.5971104633214008,
    "val/test_score/2wiki_em": 0.46484375,
    "val/test_score/popqa_f1": 0.46656323277417033,
    "val/test_score/popqa_f1_format_penalty": 0.4470319827741703,
    "val/test_score/popqa_em": 0.373046875,
    "val/test_score/tq_f1": 0.72589981387437,
    "val/test_score/tq_f1_format_penalty": 0.70636856387437,
    "val/test_score/tq_em": 0.5703125,
    "val/test_score/hotpotqa_f1": 0.4890166365752303,
    "val/test_score/hotpotqa_f1_format_penalty": 0.44995413657523037,
    "val/test_score/hotpotqa_em": 0.318359375,
    "val/test_score/Bamboogle_f1": 0.6243047619047619,
    "val/test_score/Bamboogle_f1_format_penalty": 0.6243047619047619,
    "val/test_score/Bamboogle_em": 0.48,
    "val/test_score/nq_f1": 0.4056688458646869,
    "val/test_score/nq_f1_format_penalty": 0.39001326856527396,
    "val/test_score/nq_em": 0.22504892367906065,
    "val/test_score/musique_f1": 0.23481815648104598,
    "val/test_score/musique_f1_format_penalty": 0.094193156481046,
    "val/test_score/musique_em": -0.017578125
}

**Comment by GuoqingWang1** [comment]
Thank you very much for your time and interest! I believe the main reason for the performance gap is the tool_server itself. Additionally, I noticed a significant difference between *_f1 and *_f1_format_penalty in the results you provided (especially on the Musique dataset). This suggests that the model training might not have fully converged yet. Would it be possible for you to continue training for a bit longer to see if it improves? Furthermore, we are increasingly aware of the reproducibility challenges caused by our inability to fully open-source the tool_server. Moving forward, we will work hard to release a local retrieval version that is much easier to reproduce (similar to Search-R1). I hope this reply is helpful to you😊!

**Comment by minyoungbae** [comment]
When you say the tool_server itself is different, does that mean the current web search environment has changed since the time you reported the performance?
I evaluated after 70 training steps (longer than 5-60 steps in your paper), but I’ll try training a bit longer as you suggested. Thanks for taking the time to reply.

**Comment by GuoqingWang1** [comment]
Yes, that's correct. The tool_server provided in our current codebase is just a simplified version of the interface. Unfortunately, our company's data privacy policy restricts us from fully open-sourcing the original tool_server.

**Comment by Yatogaii** [comment]
Hi, thanks for sharing the results!

May I ask whether your evaluation was run with the Search-R1 local retriever or with an online search API such as Google / Serper / Bing? I’m trying to reproduce the setup and want to make sure I’m using the same search backend.

Thanks!

**Comment by GuoqingWang1** [comment]
Hello, thank you for your attention. The evaluation results in the paper are based on the online API, while the other outcome-based RL results are copied from the [DeepResearcher](https://arxiv.org/abs/2504.03160) report, and the step reward RL results are copied from their respective original reports. Additionally, we are working on releasing the local search version of IGPO to better facilitate community research. Stay tuned!
